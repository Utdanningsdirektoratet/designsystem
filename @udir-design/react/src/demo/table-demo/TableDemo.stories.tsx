import type { Meta, StoryObj } from '@storybook/react-vite';
import { TableDemo } from './TableDemo';

const meta: Meta<typeof TableDemo> = {
  title: 'demo/Table Demo',
  component: TableDemo,
};

export default meta;
type Story = StoryObj<typeof TableDemo>;

export const TableStory: Story = {
  parameters: {
    layout: 'fullscreen',
    customStyles: {
      padding: 0,
    },
  },
  args: {
    'data-size': 'md',
    'data-color-scheme': 'auto',
  },
  render(args) {
    return <TableDemo {...args} />;
  },
};
