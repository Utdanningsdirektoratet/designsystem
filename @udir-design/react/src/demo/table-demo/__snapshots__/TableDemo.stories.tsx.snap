// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table Story 1`] = `
<div>
  <div data-size="sm">
    <fieldset>
      <legend data-weight="medium">
        Størrelse
        <code data-size="xs">
          (data-size)
        </code>
      </legend>
      <div
        role="radiogroup"
        tabindex="0"
      >
        <button
          data-variant="primary"
          type="button"
          value="sm"
          id="togglegroup-item-«r2»"
          aria-checked="true"
          aria-current="true"
          role="radio"
          name="togglegroup-name-«r0»"
          data-roving-tabindex-item="true"
          tabindex="0"
        >
          sm
        </button>
        <button
          data-variant="tertiary"
          type="button"
          value="md"
          id="togglegroup-item-«r4»"
          aria-checked="false"
          aria-current="false"
          role="radio"
          name="togglegroup-name-«r0»"
          data-roving-tabindex-item="true"
          tabindex="-1"
        >
          md
        </button>
        <button
          data-variant="tertiary"
          type="button"
          value="lg"
          id="togglegroup-item-«r6»"
          aria-checked="false"
          aria-current="false"
          role="radio"
          name="togglegroup-name-«r0»"
          data-roving-tabindex-item="true"
          tabindex="-1"
        >
          lg
        </button>
      </div>
    </fieldset>
    <fieldset>
      <legend data-weight="medium">
        Fargemodus
        <code data-size="xs">
          (data-color-scheme)
        </code>
      </legend>
      <div
        role="radiogroup"
        tabindex="0"
      >
        <button
          data-variant="tertiary"
          type="button"
          value="light"
          id="togglegroup-item-«r9»"
          aria-checked="false"
          aria-current="false"
          role="radio"
          name="togglegroup-name-«r7»"
          data-roving-tabindex-item="true"
          tabindex="-1"
        >
          light
        </button>
        <button
          data-variant="tertiary"
          type="button"
          value="dark"
          id="togglegroup-item-«rb»"
          aria-checked="false"
          aria-current="false"
          role="radio"
          name="togglegroup-name-«r7»"
          data-roving-tabindex-item="true"
          tabindex="-1"
        >
          dark
        </button>
        <button
          data-variant="primary"
          type="button"
          value="auto"
          id="togglegroup-item-«rd»"
          aria-checked="true"
          aria-current="true"
          role="radio"
          name="togglegroup-name-«r7»"
          data-roving-tabindex-item="true"
          tabindex="0"
        >
          auto
        </button>
      </div>
    </fieldset>
  </div>
  <div
    data-size="sm"
    data-color-scheme="auto"
  >
    <div>
      <div data-color="info">
        Elevlisten vil bli oppdatert ved semesterstart.
      </div>
      <h2 data-size="md">
        Elevliste
      </h2>
      <hr aria-hidden="true">
      <div>
        <div>
          <label data-weight="medium">
            Filtrer på utdanningsnivå
          </label>
          <div style="display: flex; gap: var(--ds-size-2); margin-top: var(--ds-size-2);">
            <label>
              <input
                type="checkbox"
                value="Barnehage"
                checked
              >
              Barnehage
            </label>
            <label>
              <input
                type="checkbox"
                value="Grunnskole"
                checked
              >
              Grunnskole
            </label>
            <label>
              <input
                type="checkbox"
                value="Videregående"
                checked
              >
              Videregående
            </label>
          </div>
        </div>
        <div>
          <input
            placeholder="Søk"
            aria-label="Søk"
            type="search"
            value
            name="search"
          >
          <button
            data-icon="true"
            data-variant="tertiary"
            type="reset"
            aria-label="Tøm"
          >
          </button>
        </div>
      </div>
      <div>
        <table data-border="true">
          <thead>
            <tr>
              <th>
                <div>
                  <input
                    id="checkbox-select-all"
                    aria-label="Select all"
                    type="checkbox"
                    value="all"
                    name="my-checkbox"
                  >
                </div>
              </th>
              <th aria-sort="none">
                <button type="button">
                  Navn
                </button>
              </th>
              <th aria-sort="none">
                <button type="button">
                  Utdanningsnivå
                </button>
              </th>
              <th>
                Sist endret
              </th>
              <th>
                Telefon
              </th>
              <th>
                Status
              </th>
              <th>
                Slett
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <div>
                  <input
                    id="checkbox-1"
                    aria-label="Check Lise Nordmann"
                    type="checkbox"
                    value="1"
                    name="my-checkbox"
                  >
                </div>
              </td>
              <td>
                <div>
                  <span
                    data-variant="circle"
                    role="img"
                    aria-label="Lise Nordmann"
                  >
                  </span>
                  Lise Nordmann
                </div>
              </td>
              <td>
                Barnehage
              </td>
              <td>
                01.01.2023
              </td>
              <td>
                22345678
              </td>
              <td>
                <span data-color="info">
                  Ny
                </span>
              </td>
              <td>
                <button
                  data-icon="true"
                  data-variant="tertiary"
                  type="button"
                  aria-haspopup="dialog"
                  data-color="danger"
                  aria-label="Slett"
                  popovertarget="«rg»"
                  popovertargetaction="show"
                >
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    width="1em"
                    height="1em"
                    fill="none"
                    viewbox="0 0 24 24"
                    focusable="false"
                    role="img"
                    aria-hidden="true"
                  >
                    <path
                      fill="currentColor"
                      fill-rule="evenodd"
                      d="M10 4.75c-.69 0-1.25.56-1.25 1.25v.25h6.5V6c0-.69-.56-1.25-1.25-1.25zm6.75 1.5V6A2.75 2.75 0 0 0 14 3.25h-4A2.75 2.75 0 0 0 7.25 6v.25H4.5a.75.75 0 0 0 0 1.5h.805l.876 11.384a1.75 1.75 0 0 0 1.745 1.616h8.148a1.75 1.75 0 0 0 1.745-1.616l.875-11.384h.806a.75.75 0 0 0 0-1.5h-2.75m-6 4.25a.75.75 0 0 0-1.5 0v6a.75.75 0 0 0 1.5 0zM14 9.75a.75.75 0 0 1 .75.75v6a.75.75 0 0 1-1.5 0v-6a.75.75 0 0 1 .75-.75"
                      clip-rule="evenodd"
                    >
                    </path>
                  </svg>
                </button>
                <span
                  role="tooltip"
                  id="«rg»"
                  popover="manual"
                  style="opacity: 0;"
                >
                  Slett
                </span>
                <dialog data-modal="true">
                  <button
                    data-icon="true"
                    data-variant="tertiary"
                    type="button"
                    aria-label="Lukk dialogvindu"
                    data-color="neutral"
                    data-command="close"
                  >
                  </button>
                  <div>
                    <h2 data-size="xs">
                      Er du sikker på at du vil slette Lise Nordmann?
                    </h2>
                  </div>
                  <div>
                    <button
                      data-variant="primary"
                      type="button"
                      data-color="danger"
                    >
                      Slett
                    </button>
                    <button
                      data-variant="secondary"
                      type="button"
                    >
                      Avbryt
                    </button>
                  </div>
                </dialog>
              </td>
            </tr>
            <tr>
              <td>
                <div>
                  <input
                    id="checkbox-2"
                    aria-label="Check Ola Nordmann"
                    type="checkbox"
                    value="2"
                    name="my-checkbox"
                  >
                </div>
              </td>
              <td>
                <div>
                  <span
                    data-variant="circle"
                    role="img"
                    aria-label="Ola Nordmann"
                  >
                  </span>
                  Ola Nordmann
                </div>
              </td>
              <td>
                Grunnskole
              </td>
              <td>
                01.02.2023
              </td>
              <td>
                87654321
              </td>
              <td>
                <span data-color="success">
                  Eksportert
                </span>
              </td>
              <td>
                <button
                  data-icon="true"
                  data-variant="tertiary"
                  type="button"
                  aria-haspopup="dialog"
                  data-color="danger"
                  aria-label="Slett"
                  popovertarget="«ri»"
                  popovertargetaction="show"
                >
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    width="1em"
                    height="1em"
                    fill="none"
                    viewbox="0 0 24 24"
                    focusable="false"
                    role="img"
                    aria-hidden="true"
                  >
                    <path
                      fill="currentColor"
                      fill-rule="evenodd"
                      d="M10 4.75c-.69 0-1.25.56-1.25 1.25v.25h6.5V6c0-.69-.56-1.25-1.25-1.25zm6.75 1.5V6A2.75 2.75 0 0 0 14 3.25h-4A2.75 2.75 0 0 0 7.25 6v.25H4.5a.75.75 0 0 0 0 1.5h.805l.876 11.384a1.75 1.75 0 0 0 1.745 1.616h8.148a1.75 1.75 0 0 0 1.745-1.616l.875-11.384h.806a.75.75 0 0 0 0-1.5h-2.75m-6 4.25a.75.75 0 0 0-1.5 0v6a.75.75 0 0 0 1.5 0zM14 9.75a.75.75 0 0 1 .75.75v6a.75.75 0 0 1-1.5 0v-6a.75.75 0 0 1 .75-.75"
                      clip-rule="evenodd"
                    >
                    </path>
                  </svg>
                </button>
                <span
                  role="tooltip"
                  id="«ri»"
                  popover="manual"
                  style="opacity: 0;"
                >
                  Slett
                </span>
                <dialog data-modal="true">
                  <button
                    data-icon="true"
                    data-variant="tertiary"
                    type="button"
                    aria-label="Lukk dialogvindu"
                    data-color="neutral"
                    data-command="close"
                  >
                  </button>
                  <div>
                    <h2 data-size="xs">
                      Er du sikker på at du vil slette Ola Nordmann?
                    </h2>
                  </div>
                  <div>
                    <button
                      data-variant="primary"
                      type="button"
                      data-color="danger"
                    >
                      Slett
                    </button>
                    <button
                      data-variant="secondary"
                      type="button"
                    >
                      Avbryt
                    </button>
                  </div>
                </dialog>
              </td>
            </tr>
            <tr>
              <td>
                <div>
                  <input
                    id="checkbox-3"
                    aria-label="Check Kari Nordmann"
                    type="checkbox"
                    value="3"
                    name="my-checkbox"
                  >
                </div>
              </td>
              <td>
                <div>
                  <span
                    data-variant="circle"
                    role="img"
                    aria-label="Kari Nordmann"
                  >
                  </span>
                  Kari Nordmann
                </div>
              </td>
              <td>
                Videregående
              </td>
              <td>
                01.03.2023
              </td>
              <td>
                11223344
              </td>
              <td>
                <span data-color="warning">
                  Ikke eksportert
                </span>
              </td>
              <td>
                <button
                  data-icon="true"
                  data-variant="tertiary"
                  type="button"
                  aria-haspopup="dialog"
                  data-color="danger"
                  aria-label="Slett"
                  popovertarget="«rk»"
                  popovertargetaction="show"
                >
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    width="1em"
                    height="1em"
                    fill="none"
                    viewbox="0 0 24 24"
                    focusable="false"
                    role="img"
                    aria-hidden="true"
                  >
                    <path
                      fill="currentColor"
                      fill-rule="evenodd"
                      d="M10 4.75c-.69 0-1.25.56-1.25 1.25v.25h6.5V6c0-.69-.56-1.25-1.25-1.25zm6.75 1.5V6A2.75 2.75 0 0 0 14 3.25h-4A2.75 2.75 0 0 0 7.25 6v.25H4.5a.75.75 0 0 0 0 1.5h.805l.876 11.384a1.75 1.75 0 0 0 1.745 1.616h8.148a1.75 1.75 0 0 0 1.745-1.616l.875-11.384h.806a.75.75 0 0 0 0-1.5h-2.75m-6 4.25a.75.75 0 0 0-1.5 0v6a.75.75 0 0 0 1.5 0zM14 9.75a.75.75 0 0 1 .75.75v6a.75.75 0 0 1-1.5 0v-6a.75.75 0 0 1 .75-.75"
                      clip-rule="evenodd"
                    >
                    </path>
                  </svg>
                </button>
                <span
                  role="tooltip"
                  id="«rk»"
                  popover="manual"
                  style="opacity: 0;"
                >
                  Slett
                </span>
                <dialog data-modal="true">
                  <button
                    data-icon="true"
                    data-variant="tertiary"
                    type="button"
                    aria-label="Lukk dialogvindu"
                    data-color="neutral"
                    data-command="close"
                  >
                  </button>
                  <div>
                    <h2 data-size="xs">
                      Er du sikker på at du vil slette Kari Nordmann?
                    </h2>
                  </div>
                  <div>
                    <button
                      data-variant="primary"
                      type="button"
                      data-color="danger"
                    >
                      Slett
                    </button>
                    <button
                      data-variant="secondary"
                      type="button"
                    >
                      Avbryt
                    </button>
                  </div>
                </dialog>
              </td>
            </tr>
            <tr>
              <td>
                <div>
                  <input
                    id="checkbox-4"
                    aria-label="Check Per Nordmann"
                    type="checkbox"
                    value="4"
                    name="my-checkbox"
                  >
                </div>
              </td>
              <td>
                <div>
                  <span
                    data-variant="circle"
                    role="img"
                    aria-label="Per Nordmann"
                  >
                  </span>
                  Per Nordmann
                </div>
              </td>
              <td>
                Barnehage
              </td>
              <td>
                01.04.2023
              </td>
              <td>
                44332211
              </td>
              <td>
                <span data-color="info">
                  Ny
                </span>
              </td>
              <td>
                <button
                  data-icon="true"
                  data-variant="tertiary"
                  type="button"
                  aria-haspopup="dialog"
                  data-color="danger"
                  aria-label="Slett"
                  popovertarget="«rm»"
                  popovertargetaction="show"
                >
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    width="1em"
                    height="1em"
                    fill="none"
                    viewbox="0 0 24 24"
                    focusable="false"
                    role="img"
                    aria-hidden="true"
                  >
                    <path
                      fill="currentColor"
                      fill-rule="evenodd"
                      d="M10 4.75c-.69 0-1.25.56-1.25 1.25v.25h6.5V6c0-.69-.56-1.25-1.25-1.25zm6.75 1.5V6A2.75 2.75 0 0 0 14 3.25h-4A2.75 2.75 0 0 0 7.25 6v.25H4.5a.75.75 0 0 0 0 1.5h.805l.876 11.384a1.75 1.75 0 0 0 1.745 1.616h8.148a1.75 1.75 0 0 0 1.745-1.616l.875-11.384h.806a.75.75 0 0 0 0-1.5h-2.75m-6 4.25a.75.75 0 0 0-1.5 0v6a.75.75 0 0 0 1.5 0zM14 9.75a.75.75 0 0 1 .75.75v6a.75.75 0 0 1-1.5 0v-6a.75.75 0 0 1 .75-.75"
                      clip-rule="evenodd"
                    >
                    </path>
                  </svg>
                </button>
                <span
                  role="tooltip"
                  id="«rm»"
                  popover="manual"
                  style="opacity: 0;"
                >
                  Slett
                </span>
                <dialog data-modal="true">
                  <button
                    data-icon="true"
                    data-variant="tertiary"
                    type="button"
                    aria-label="Lukk dialogvindu"
                    data-color="neutral"
                    data-command="close"
                  >
                  </button>
                  <div>
                    <h2 data-size="xs">
                      Er du sikker på at du vil slette Per Nordmann?
                    </h2>
                  </div>
                  <div>
                    <button
                      data-variant="primary"
                      type="button"
                      data-color="danger"
                    >
                      Slett
                    </button>
                    <button
                      data-variant="secondary"
                      type="button"
                    >
                      Avbryt
                    </button>
                  </div>
                </dialog>
              </td>
            </tr>
            <tr>
              <td>
                <div>
                  <input
                    id="checkbox-5"
                    aria-label="Check Anne Nordmann"
                    type="checkbox"
                    value="5"
                    name="my-checkbox"
                  >
                </div>
              </td>
              <td>
                <div>
                  <span
                    data-variant="circle"
                    role="img"
                    aria-label="Anne Nordmann"
                  >
                  </span>
                  Anne Nordmann
                </div>
              </td>
              <td>
                Grunnskole
              </td>
              <td>
                01.05.2023
              </td>
              <td>
                55667788
              </td>
              <td>
                <span data-color="success">
                  Eksportert
                </span>
              </td>
              <td>
                <button
                  data-icon="true"
                  data-variant="tertiary"
                  type="button"
                  aria-haspopup="dialog"
                  data-color="danger"
                  aria-label="Slett"
                  popovertarget="«ro»"
                  popovertargetaction="show"
                >
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    width="1em"
                    height="1em"
                    fill="none"
                    viewbox="0 0 24 24"
                    focusable="false"
                    role="img"
                    aria-hidden="true"
                  >
                    <path
                      fill="currentColor"
                      fill-rule="evenodd"
                      d="M10 4.75c-.69 0-1.25.56-1.25 1.25v.25h6.5V6c0-.69-.56-1.25-1.25-1.25zm6.75 1.5V6A2.75 2.75 0 0 0 14 3.25h-4A2.75 2.75 0 0 0 7.25 6v.25H4.5a.75.75 0 0 0 0 1.5h.805l.876 11.384a1.75 1.75 0 0 0 1.745 1.616h8.148a1.75 1.75 0 0 0 1.745-1.616l.875-11.384h.806a.75.75 0 0 0 0-1.5h-2.75m-6 4.25a.75.75 0 0 0-1.5 0v6a.75.75 0 0 0 1.5 0zM14 9.75a.75.75 0 0 1 .75.75v6a.75.75 0 0 1-1.5 0v-6a.75.75 0 0 1 .75-.75"
                      clip-rule="evenodd"
                    >
                    </path>
                  </svg>
                </button>
                <span
                  role="tooltip"
                  id="«ro»"
                  popover="manual"
                  style="opacity: 0;"
                >
                  Slett
                </span>
                <dialog data-modal="true">
                  <button
                    data-icon="true"
                    data-variant="tertiary"
                    type="button"
                    aria-label="Lukk dialogvindu"
                    data-color="neutral"
                    data-command="close"
                  >
                  </button>
                  <div>
                    <h2 data-size="xs">
                      Er du sikker på at du vil slette Anne Nordmann?
                    </h2>
                  </div>
                  <div>
                    <button
                      data-variant="primary"
                      type="button"
                      data-color="danger"
                    >
                      Slett
                    </button>
                    <button
                      data-variant="secondary"
                      type="button"
                    >
                      Avbryt
                    </button>
                  </div>
                </dialog>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div>
        <nav aria-label="Sidenavigering">
          <ul>
            <li>
              <button
                aria-disabled="true"
                data-variant="tertiary"
                type="button"
              >
                Forrige
              </button>
            </li>
            <li>
              <button
                data-variant="primary"
                type="button"
                aria-current="page"
                aria-label="Side 1"
              >
                1
              </button>
            </li>
            <li>
              <button
                data-variant="tertiary"
                type="button"
                aria-label="Side 2"
              >
                2
              </button>
            </li>
            <li>
              <button
                data-variant="tertiary"
                type="button"
                aria-label="Side 3"
              >
                3
              </button>
            </li>
            <li>
              <button
                aria-disabled="false"
                data-variant="tertiary"
                type="button"
              >
                Neste
              </button>
            </li>
          </ul>
        </nav>
        <button
          data-variant="primary"
          type="button"
          aria-haspopup="dialog"
        >
          Eksporter liste
        </button>
        <dialog data-modal="true">
          <button
            data-icon="true"
            data-variant="tertiary"
            type="button"
            aria-label="Lukk dialogvindu"
            data-color="neutral"
            data-command="close"
          >
          </button>
          <div>
            <h2 data-size="xs">
              Eksporter liste
            </h2>
            <p data-variant="default">
              I hvilket format ønsker du å eksportere listen?
            </p>
            <div>
              <button
                data-variant="secondary"
                type="button"
              >
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  width="1em"
                  height="1em"
                  fill="none"
                  viewbox="0 0 24 24"
                  focusable="false"
                  role="img"
                  aria-hidden="true"
                >
                  <path
                    fill="currentColor"
                    fill-rule="evenodd"
                    d="M13.25 3.5a.25.25 0 0 0-.25-.25H6.5c-.69 0-1.25.56-1.25 1.25v6c0 .138.112.25.25.25h13a.25.25 0 0 0 .25-.25V9a.25.25 0 0 0-.25-.25h-4c-.69 0-1.25-.56-1.25-1.25zM15 7.25a.25.25 0 0 1-.25-.25V4.293a.25.25 0 0 1 .427-.177l2.707 2.707a.25.25 0 0 1-.177.427zm-12 5a.75.75 0 0 0-.75.75v7c0 .414.336.75.75.75h18a.75.75 0 0 0 .75-.75v-7a.75.75 0 0 0-.75-.75zm4.7 1.64q.54 0 .938.203.405.203.623.567.224.364.224.854 0 .483-.224.854-.224.364-.623.567a2.04 2.04 0 0 1-.938.203h-.693V19h-1.05v-5.11zm4.302 0q.533 0 .924.203.4.203.616.574.224.365.224.86v1.828q0 .49-.224.86-.216.372-.616.582-.391.203-.924.203h-1.659v-5.11zm2.798 0V19h1.05v-2.016h2.058v-.98h-2.072V14.87h2.254v-.98zm-6.568 1.134q-.196-.19-.532-.19h-.693v1.359H7.7q.336 0 .532-.182a.65.65 0 0 0 .203-.497.63.63 0 0 0-.203-.49m4.288.028a.73.73 0 0 0-.518-.182h-.609v3.15h.61q.322 0 .517-.182a.63.63 0 0 0 .196-.483v-1.827a.62.62 0 0 0-.196-.476"
                    clip-rule="evenodd"
                  >
                  </path>
                </svg>
                PDF
              </button>
              <button
                data-variant="secondary"
                type="button"
              >
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  width="1em"
                  height="1em"
                  fill="none"
                  viewbox="0 0 24 24"
                  focusable="false"
                  role="img"
                  aria-hidden="true"
                >
                  <path
                    fill="currentColor"
                    fill-rule="evenodd"
                    d="M13.25 3.5a.25.25 0 0 0-.25-.25H6.5c-.69 0-1.25.56-1.25 1.25v6c0 .138.112.25.25.25h13a.25.25 0 0 0 .25-.25V9a.25.25 0 0 0-.25-.25h-4c-.69 0-1.25-.56-1.25-1.25zM15 7.25a.25.25 0 0 1-.25-.25V4.293a.25.25 0 0 1 .427-.177l2.707 2.707a.25.25 0 0 1-.177.427zm-12 5a.75.75 0 0 0-.75.75v7c0 .414.336.75.75.75h18a.75.75 0 0 0 .75-.75v-7a.75.75 0 0 0-.75-.75zm4.622 6.82q-.511 0-.896-.189a1.5 1.5 0 0 1-.602-.539 1.55 1.55 0 0 1-.21-.812v-2.17q0-.469.21-.812.217-.343.602-.532a1.95 1.95 0 0 1 .896-.196q.518 0 .896.196.385.189.595.532.217.344.217.812H8.28q0-.308-.175-.469-.168-.16-.483-.16t-.49.16q-.168.16-.168.47v2.17q0 .3.168.468.175.16.49.161.315 0 .483-.16.175-.17.175-.47h1.05q0 .462-.217.812-.21.343-.595.54-.378.188-.896.188m4.38 0q-.54 0-.939-.182a1.43 1.43 0 0 1-.616-.51 1.4 1.4 0 0 1-.217-.778h1.05q0 .259.196.413.204.147.546.147.33 0 .511-.147a.49.49 0 0 0 .19-.406.57.57 0 0 0-.14-.385.7.7 0 0 0-.393-.217l-.518-.119q-.65-.155-1.015-.567a1.53 1.53 0 0 1-.357-1.022q0-.448.203-.777.21-.336.588-.518t.896-.182q.784 0 1.24.392.46.385.461 1.043h-1.05a.47.47 0 0 0-.175-.385q-.168-.14-.49-.14-.3 0-.462.14-.16.133-.16.385 0 .225.125.385a.68.68 0 0 0 .371.21l.546.126q.68.155 1.03.56.35.4.35 1.022 0 .448-.225.791-.216.343-.616.532a2.1 2.1 0 0 1-.93.19m2.482-5.18L15.75 19h1.33l1.253-5.11h-1.057l-.686 3.22q-.036.175-.077.434-.043.26-.07.441a8 8 0 0 0-.084-.44q-.05-.26-.084-.428l-.707-3.227z"
                    clip-rule="evenodd"
                  >
                  </path>
                </svg>
                CSV
              </button>
            </div>
          </div>
        </dialog>
      </div>
    </div>
  </div>
</div>
`;
